lib/improver/cli/clize_routines.py

```diff
@@ -0,0 +1,206 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Routines for clize to run."""
+import clize
+import clize.help
+import sigtools.wrappers
+
+# help helpers
+
+
+def docutilize(obj):
+    """
+
+    Args:
+        obj (str or obj):
+            Takes an object and changes it's docstrings to a suitable format
+            for clize.
+    Returns:
+        (str or obj):
+            A string with replaced docstrings. or an altered string depending
+            on the format of the input.
+    """
+    from inspect import cleandoc
+    from sphinx.ext.napoleon.docstring import GoogleDocstring, NumpyDocstring
+    if isinstance(obj, str):
+        doc = obj
+    else:
+        doc = obj.__doc__
+    doc = cleandoc(doc)
+    doc = str(NumpyDocstring(doc))
+    doc = str(GoogleDocstring(doc))
+    # exception and keyword markup seems to trip up the docutils parser
+    doc = doc.replace(':exc:', '')
+    doc = doc.replace(':keyword', ':param')
+    doc = doc.replace(':kwtype', ':type')
+    if isinstance(obj, str):
+        return doc
+    obj.__doc__ = doc
+    return obj
+
+
+class HelpForNapoleonDocstring(clize.help.HelpForAutodetectedDocstring):
+    """Help for Napolean Docstrings."""
+    def add_docstring(self, docstring, *args, **kwargs):
+        """Adds the updated docstring."""
+        docstring = docutilize(docstring)
+        super().add_docstring(docstring, *args, **kwargs)
+
+
+class DocutilizeClizeHelp(clize.help.ClizeHelp):
+    """Subclass to build Napoleon docstring from subject."""
+    def __init__(self, subject, owner,
+                 builder=HelpForNapoleonDocstring.from_subject):
+        super().__init__(subject, owner, builder)
+
+# converters
+
+
+def maybe_coerce_with(convert, obj, **kwargs):
+    """Apply converter if str, pass through otherwise."""
+    return convert(obj, **kwargs) if isinstance(obj, str) else obj
+
+
+@clize.parser.value_converter
+def inputcube(to_convert):
+    """
+
+    Args:
+        to_convert (string or obj):
+            calls maybe_coerce_with function with the input and load_cube.
+
+    Returns:
+        (obj):
+            The result of maybe_coerce_with.
+
+    """
+    from improver.utilities.load import load_cube
+    return maybe_coerce_with(load_cube, to_convert)
+
+
+@clize.parser.value_converter
+def optionalcube(to_convert):
+    """
+
+    Args:
+        to_convert (string or obj):
+            calls maybe_coerce_with function with the input and load_cube.
+
+    Returns:
+        (obj):
+            The result of maybe_coerce_with.
+
+    """
+    from improver.utilities.load import load_cube
+    return maybe_coerce_with(load_cube, to_convert, **{'allow_none': True})
```

Comment:
Note that `maybe_coerce_with` applies conversion only on strings, so you don't really need this `optionalcube` special case. If `to_convert` is `None` it will be returned as is.

Reply:
Yeah, fair enough.